service/src/main/java/bio/terra/pipelines/app/controller/JobsApiController.java

```diff
@@ -61,93 +49,23 @@ private SamUser getAuthenticatedInfo() {
   // -- Jobs --
 
   @Override
-  public ResponseEntity<ApiPostJobResponse> createJob(
-      @PathVariable("pipelineId") String pipelineId, @RequestBody ApiPostJobRequestBody body) {
+  public ResponseEntity<ApiJobReport> getJob(@PathVariable("jobId") UUID jobId) {
     final SamUser userRequest = getAuthenticatedInfo();
     String userId = userRequest.getSubjectId();
-    String pipelineVersion = body.getPipelineVersion();
-    Object pipelineInputs = body.getPipelineInputs();
-
-    if (!pipelinesService.pipelineExists(pipelineId)) {
-      throw new PipelineNotFoundException(
-          String.format("Requested pipeline %s not found.", pipelineId));
-    }
-
-    logger.info(
-        "Creating {} pipeline job (version {}) for {} user {} with inputs {}",
-        pipelineId,
-        pipelineVersion,
-        userRequest.getEmail(),
-        userId,
-        pipelineInputs);
-
-    // TODO assuming we will write outputs back to source workspace, we will need to check user
-    // permissions for write access to the workspace - explore interceptors
-
-    UUID createdJobUuid =
-        jobsService.createJob(userId, pipelineId, pipelineVersion, pipelineInputs);
-    if (createdJobUuid == null) {
-      logger.error("New {} pipeline job creation failed.", pipelineId);
-      throw new ApiException("An internal error occurred.");
-    }
-
-    // eventually we'll expand this out to kick off the imputation pipeline flight but for
-    // now this is good enough.
-    imputationService.queryForWorkspaceApps();
-
-    ApiPostJobResponse createdJobResponse = new ApiPostJobResponse();
-    createdJobResponse.setJobId(createdJobUuid.toString());
-    logger.info("Created {} job {}", pipelineId, createdJobUuid);
-    MetricsUtils.incrementPipelineRun(pipelineId);
-
-    return new ResponseEntity<>(createdJobResponse, HttpStatus.OK);
-  }
-
-  @Override
-  public ResponseEntity<ApiGetJobResponse> getJob(
-      @PathVariable("pipelineId") String pipelineId, @PathVariable("jobId") UUID jobId) {
-    final SamUser userRequest = getAuthenticatedInfo();
-    String userId = userRequest.getSubjectId();
-    Job job = jobsService.getJob(userId, pipelineId, jobId);
-    ApiGetJobResponse result = jobToApi(job);
-
+    logger.info("Retrieving jobId {} for userId {}", jobId, userId);
+    FlightState flightState = stairwayJobService.retrieveJob(jobId, userId);
```

Comment:
how are we making sure that we are correctly handling 403s (dont have permission to see job id being asked for) and 404s (job id doesnt exist)

Reply:
discussed in Slack - had been overly cautious with returning 404s instead of 403s, but this isn't necessary. should be returning 403s properly now.